features/release.feature

```diff
@@ -367,6 +367,36 @@ MARKDOWN
           ASSET_TARBALL
           """
 
+  Scenario: Delete a release
+    Given the GitHub API server:
+      """
+      get('/repos/mislav/will_paginate/releases') {
+          json [
+            { url: 'https://api.github.com/repos/mislav/will_paginate/releases/123',
+              upload_url: 'https://api.github.com/uploads/assets{?name,label}',
+              tag_name: 'v1.2.0',
+              name: 'will_paginate 1.2.0',
+              draft: true,
+              prerelease: false,
+              assets: [
+                { url: 'https://api.github.com/repos/mislav/will_paginate/assets/9876',
+                  name: 'hello-1.2.0.tar.gz',
+                },
+              ],
+            },
+          ]
+      }
+
+      delete('/repos/mislav/will_paginate/releases/123') {
+        status 204
+      }
+      """
+    When I successfully run `hub release delete v1.2.0`
+    Then the output should contain exactly:
+      """
+      Deleting release related to v1.2.0...\n
```

Comment:
I like the idea, though I'm obviously not as familiar with the implementation as you. I'm not sure I'd be able to whip up something more complex then the simple check for the status code. I'm usually a C developer, not so much web APIs.

If you really want to see this, I can try to figure it out, but I'd be just as happy if it just showed up. It seems like the `204` return code should be enough to validate that it worked. I'll have to verify this though.